src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -1624,90 +1615,8 @@ because xbuild doesn't support framework reference assemblies.
-		UseShortFileNames="$(UseShortFileNames)"
-		JavaPlatformJarPath="$(JavaPlatformJarPath)"
-		ManifestFiles="$(ResgenTemporaryDirectory)\AndroidManifest.xml"
-		PackageName="$(_AndroidPackage)"
-		ApplicationName="$(_AndroidPackage)"
-		ResourceDirectory="$(MonoAndroidResDirIntermediate)"
-		JavaDesignerOutputDirectory="$(ResgenTemporaryDirectory)"
-		ResourceOutputFile="$(ResgenTemporaryDirectory)\resources.apk"
-		ExtraPackages="$(AaptExtraPackages)"
-		AdditionalResourceDirectories="@(LibraryResourceDirectories)"
-		LibraryProjectJars="@(ExtractedJarImports)"
-		ExtraArgs="$(AndroidResgenExtraArgs)"
-		ToolPath="$(AaptToolPath)"
-		ToolExe="$(AaptToolExe)"
-		AdditionalAndroidResourcePaths="@(_AdditonalAndroidResourceCachePaths)"
-		ApiLevel="$(_AndroidTargetSdkVersion)"
-		AndroidUseLatestPlatformSdk="$(AndroidUseLatestPlatformSdk)"
-		ResourceNameCaseMap="$(_AndroidResourceNameCaseMap)"
-		AssemblyIdentityMapFile="$(_AndroidLibrayProjectAssemblyMapFile)"
-		YieldDuringToolExecution="$(YieldDuringToolExecution)"
-		ExplicitCrunch="$(AndroidExplicitCrunch)"
-		SupportedAbis="$(_BuildTargetAbis)"
-		ResourceSymbolsTextFileDirectory="$(IntermediateOutputPath)"
-		ContinueOnError="$(DesignTimeBuild)"
-	/>
-
-	<Aapt2Link
-		Condition="'$(_AndroidResourceDesignerFile)' != '' And '$(_AndroidUseAapt2)' == 'True' And Exists ('$(_AndroidLibraryFlatArchivesDirectory)\compiled.flata')"
-		ContinueOnError="$(DesignTimeBuild)"
-		ResourceNameCaseMap="$(_AndroidResourceNameCaseMap)"
-		AssemblyIdentityMapFile="$(_AndroidLibrayProjectAssemblyMapFile)"
-		ImportsDirectory="$(_LibraryProjectImportsDirectoryName)"
-		UseShortFileNames="$(UseShortFileNames)"
-		OutputImportDirectory="$(_AndroidLibrayProjectIntermediatePath)"
-		OutputFile="$(ResgenTemporaryDirectory)\resources.apk"
-		PackageName="$(_AndroidPackage)"
-		ApplicationName="$(_AndroidPackage)"
-		JavaPlatformJarPath="$(JavaPlatformJarPath)"
-		JavaDesignerOutputDirectory="$(ResgenTemporaryDirectory)"
-		CompiledResourceFlatArchive="$(_AndroidLibraryFlatArchivesDirectory)\compiled.flata"
-		ManifestFiles="$(ResgenTemporaryDirectory)\AndroidManifest.xml"
-		AdditionalResourceArchives="@(_LibraryResourceHashDirectories->'$(_AndroidLibraryFlatArchivesDirectory)%(Hash).flata')"
-		AdditionalAndroidResourcePaths="@(_LibraryResourceHashDirectories)"
-		YieldDuringToolExecution="$(YieldDuringToolExecution)"
-		ResourceSymbolsTextFile="$(IntermediateOutputPath)R.txt"
-		ResourceDirectories="$(MonoAndroidResDirIntermediate)"
-		ExtraArgs="$(AndroidAapt2LinkExtraArgs)"
-		ToolPath="$(Aapt2ToolPath)"
-		ToolExe="$(Aapt2ToolExe)"
-	/>
-	<ItemGroup>
-		<FileWrites Include="$(IntermediateOutputPath)R.txt" Condition=" '$(_AndroidUseAapt2)' == 'True' And Exists ('$(IntermediateOutputPath)R.txt') " />
-	</ItemGroup>
+	<CallTarget Targets="_UpdateAndroidResgenAapt" Condition="'$(_AndroidUseAapt2)' != 'True'" />
+	<CallTarget Targets="_UpdateAndroidResgenAapt2" Condition="'$(_AndroidUseAapt2)' == 'True'" />
```

Comment:
You might see if these two can just go in the `DependsOnTargets` list?

I think they will get skipped if the `Inputs` and `Outputs` cause `_UpdateAndroidResgen` to skip? But I might be thinking of a `Condition`...

Reply:
That was the whole point of using CallTarget here. I only want these to run IF the _UpdateAndroidResgen target runs.

Reply:
That said with the changes I had to make for `_PrepareUpdateAndroidResgen` it might be better to switch this out a bit more.

Reply:
Yeah I tested it, and you can't:
```xml
<Project DefaultTargets="C">
  <Target Name="A">
    <Message Text="A" />
  </Target>
  <Target Name="B">
    <Message Text="B" />
  </Target>
  <Target Name="C" Inputs="$(MSBuildThisFile)" Outputs="C.stamp" DependsOnTargets="A;B">
    <Message Text="C" />
    <Touch Files="C.stamp" AlwaysCreate="True" />
  </Target>
</Project>
```
`A` and `B` run if `C` is skipped.

But if you put a `Condition` on `C` instead of `Inputs` and `Outputs`, `A` and `B` will be skipped.

Just a little confusing... the two cases.